src/sim_engine/filesystem.rs

```diff
@@ -47,4 +50,9 @@ impl Filesystem for SimFilesystem {
         self.name = String::from(new_name);
         Ok(())
     }
+
+    fn add_ancestor(&mut self, parent: Uuid) -> EngineResult<()> {
```

Comment:
Since this is a private function, and ```push()``` can only panic, might as well have it return ```()``` rather than an ```EngineResult``` for now.